Moonlight macOS/ViewController/ViewController.m

```diff
@@ -0,0 +1,245 @@
+}
+- (void)viewDidAppear {
+    [super viewDidAppear];
+    
+    if (self.view.bounds.size.height == NSScreen.mainScreen.frame.size.height && self.view.bounds.size.width == NSScreen.mainScreen.frame.size.width)
+    {
+        [self.view.window toggleFullScreen:self];
+        [self.view.window setStyleMask:[self.view.window styleMask] & ~NSWindowStyleMaskResizable];
+        return;
+    }
+    [_buttonLaunch setEnabled:false];
+    [_popupButtonSelection removeAllItems];
+    _settingsView = [self.childViewControllers lastObject];
+    
+    if (_settingsView.getCurrentHost != nil)
+        _textFieldHost.stringValue = _settingsView.getCurrentHost;
+    settingsFrameHeight = _layoutConstraintSetupFrame.constant;
+    _layoutConstraintSetupFrame.constant = 0;
+    showSettings = false;
+    
+    if (error != 0) {
+        [self showAlert:[NSString stringWithFormat: @"The connection terminated."]];
+    }
+}
+
+-(void) showAlert:(NSString*) message {
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        _alert = [NSAlert new];
+        _alert.messageText = message;
+        [_alert beginSheetModalForWindow:[self.view window] completionHandler:^(NSInteger result) {
+            NSLog(@"Success");
+        }];
+    });
+}
+
+
+- (void)setRepresentedObject:(id)representedObject {
+    [super setRepresentedObject:representedObject];
+
+    // Update the view, if already loaded.
+}
+
+- (void) saveSettings {
+    DataManager* dataMan = [[DataManager alloc] init];
+    NSInteger framerate = [_settingsView getChosenFrameRate];
+    NSInteger height = [_settingsView getChosenStreamHeight];
+    NSInteger width = [_settingsView getChosenStreamWidth];
+    NSInteger streamingRemotely = [_settingsView getRemoteOptions];
+    NSInteger bitrate = [_settingsView getChosenBitrate];
+    [dataMan saveSettingsWithBitrate:bitrate framerate:framerate height:height width:width
+                    onscreenControls:0 remote:streamingRemotely];
+}
+
+
+- (void)controlTextDidChange:(NSNotification *)obj {
+    
+}
+
+- (IBAction)buttonLaunchPressed:(id)sender {
+    [self saveSettings];
+    DataManager* dataMan = [[DataManager alloc] init];
+    TemporarySettings* streamSettings = [dataMan getSettings];
+    _streamConfig = [[StreamConfiguration alloc] init];
+    _streamConfig.frameRate = [streamSettings.framerate intValue];
+    _streamConfig.bitRate = [streamSettings.bitrate intValue];
+    _streamConfig.height = [streamSettings.height intValue];
+    _streamConfig.width = [streamSettings.width intValue];
+    _streamConfig.streamingRemotely = [streamSettings.streamingRemotely intValue];
+    _streamConfig.host = _textFieldHost.stringValue;
+    _streamConfig.appID = [_sortedAppList[_popupButtonSelection.indexOfSelectedItem] id];
+    [self transitionToStreamView];
+}
+
+- (IBAction)textFieldAction:(id)sender {
+    [self buttonConnectPressed:self];
+}
+
+- (IBAction)buttonConnectPressed:(id)sender {
+    _host = _textFieldHost.stringValue;
```

Comment:
Can we share some of these functions in this class with iOS?

Reply:
Yes, but not as smooth as in most of the other classes. We'd have to create a helper class that both view controllers can call. Or are there better ways?